docs(README): Update README with the Troubleshooting section (#36)
aquasecurity/harbor-scanner-aqua: README.md

## TOC

- [Requirements](#requirements)
- [How does it work?](#how-does-it-work)
- [Getting started](#getting-started)
  - [Prerequisites](#prerequisites)
  - [Build](#build)

>   - [Running on Kubernetes](#running-on-kubernetes)

Reviewer: Should this be "Running locally on Kubernetes" or "Running on Kubernetes on your laptop" to distinguish from the Kubernetes section under "Deployment"?

Author: It's under the Getting started section, i.e. local development to distinguish it from the Deployment section.

   If you're not using Kubernetes to run the adapter service, you have to download the `scannercli` executable from the
   Aqua downloads page manually and mount it at `/usr/local/bin/scannercli`.
   See [Aqua Scanner Executable Binary][aqua-docs-scanner-binary] for more details on manual download.

> 4. It is highly recommended to create a new user in the Aqua CSP management console with credentials dedicated to the
>    Harbor adapter, e.g. `harbor_scanner`. The adapter does not require the full access to the management console.
>    Therefore, only the `Scanner` role should be assigned to such user. The `Scanner` role is the only permission
>    required by the `scannercli` executable binary which is run by the adapter service on each scan request.

Reviewer: ```suggestion
4. It is highly recommended to create a new user in the Aqua CSP management console with credentials dedicated to the
   Harbor adapter, e.g. `harbor_scanner`. The adapter does not need full access to Aqua: the `Scanner` role is the only permission
   required for the `scannercli` executable binary which is run by the adapter service on each scan request.
   Therefore, create your `harbor_scanner` user and assign it only the `Scanner` role. 
```

   ![](docs/images/aqua_user_for_harbor.png)

> 5. It is also highly recommended to create a new user in Harbor with the only permission to pull images by the Aqua CSP
>    scanner, e.g. `aqua_scanner`. Please remember to add also such user as a member of the project that you intend to
>    scan the images from.

Reviewer: ```suggestion
5. It is also highly recommended to create a new user in Harbor for the Aqua CSP scanner, with permission only to pull images from Harbor, e.g. `aqua_scanner`. Please remember to add this user as a member of the project in Harbor that you intend to
   scan the images from.
```

    library/mongo:3.4-xenial
```

> Finally, the output report is transformed to Harbor model and display in Harbor UI.

Reviewer: ```suggestion
Finally, the output report is transformed to Harbor's model and displayed in Harbor UI.
```

* [Go (version 1.13+)](https://golang.org/doc/devel/release.html#go1.13)
* Docker
* Harbor >= 1.10
* Aqua CSP >= 4.5

### Build

Run `make` to build the binary in `./scanner-adapter`:

```
$ make
```

To build into a Docker container:

```

> $ make build-image

Reviewer: Did you change this to docker-build?

Author: Yup. Good catch. Fixed!

| `SCANNER_STORE_REDIS_POOL_MAX_IDLE`   | `5`  | The max number of idle connections in the pool for a redis store      |
| `SCANNER_STORE_REDIS_SCAN_JOB_TTL`    | `1h` | The time to live for persisting scan jobs and associated scan reports |

## Troubleshooting

### Error: failed getting image manifest: 412 Precondition Failed

> Currently, there's a limitation of `scannercli` which does not accept Harbor robot account credentials passed by a
> Harbor scan job to the adapter service. This effectively means that the Aqua CSP scanner is using the credentials
> provided in Aqua CSP management console under the **Integrations** / **Image Registries** section. However, these
> credentials do not have enough permissions to bypass deployment security checker when it's enabled in the Harbor
> project configuration.

Reviewer: ```suggestion
Currently, there's a limitation of `scannercli` which does not accept Harbor robot account credentials passed by a
Harbor scan job to the adapter service. This effectively means that the Aqua CSP scanner is using the credentials
provided in Aqua CSP management console under the **Integrations** / **Image Registries** section. However, these
credentials do not have enough permissions to bypass the deployment security checker when it's enabled in the Harbor
project configuration. In other words, the deployment security checker prevents the Aqua CSP scanner from pulling an image, which it needs to be able to do in order to scan it.  
```
